editors/vscode/src/common.ts

```diff
@@ -0,0 +1,201 @@
+
+            localResourceRoots: [
+                this._extensionUri
+            ]
+        };
+
+        webviewView.webview.html = this._getHtmlForWebview(webviewView.webview);
+
+        webviewView.webview.onDidReceiveMessage(data => {
+            switch (data.command) {
+                case 'property_clicked':
+                    if (vscode.window.activeTextEditor) {
+                        const p = data.property as Property;
+                        if (p.defined_at && p.defined_at.property_definition_range) {
+                            let range = client.protocol2CodeConverter.asRange(p.defined_at.property_definition_range);
+                            vscode.window.activeTextEditor.revealRange(range);
+                            vscode.window.activeTextEditor.selection = new vscode.Selection(range.start, range.end);
+                        }
+                    }
+                    break;
+                case 'change_property':
+                    if (vscode.window.activeTextEditor) {
+                        const p = data.property as Property;
+                        if (p.defined_at && p.defined_at.expression_range) {
+                            let range = client.protocol2CodeConverter.asRange(p.defined_at.expression_range);
+                            let old = vscode.window.activeTextEditor.document.getText(range);
+                            console.log("maybe", old, data.old_value)
+                            if (old === data.old_value) {
+                                vscode.window.activeTextEditor.edit(b => b.replace(range, data.new_value));
+                            }
+                        }
+                    }
+                    break;
+            }
+        });
+
+
+        vscode.window.onDidChangeTextEditorSelection(async (event: vscode.TextEditorSelectionChangeEvent) => {
+            //client.comma sendRequest("slint/showPreview", ae.document.uri.fsPath.toString());
+
+            if (event.selections.length == 0) {
+                return;
+            }
+            let selection = event.selections[0];
+
+            let r = await vscode.commands.executeCommand(
+                "queryProperties",
+                event.textEditor.document.uri.toString(),
+                selection.active.line,
+                selection.active.character,
+            );
+            const result = r as PropertyQuery;
+            const result_str = JSON.stringify(result);
+            const msg = {
+                command: "set_properties",
+                properties: r,
+                code: event.textEditor.document.getText(),
+            };
+            webviewView.webview.postMessage(msg);
+        });
+    }
+
+    private _getHtmlForWebview(webview: vscode.Webview) {
+
+        const scriptUri = webview.asWebviewUri(vscode.Uri.joinPath(this._extensionUri, 'out/propertiesView.js'));
+        const nonce = getNonce();
+
+
+        // FIXME: share with the online editor?
```

Comment:
Not part of this patch, but just thinking out loud: Perhaps we can use a bundler to bundle `colors.css` (inline), the shared `properties.ts`, `propertiesView.ts` and a `.html` template (as inline below) into one file and also get rid of the nonce.

Reply:
Ideally, we should somehow share the .css between the online editor and this.